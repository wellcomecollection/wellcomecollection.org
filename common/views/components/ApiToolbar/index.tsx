import { FunctionComponent, useState } from 'react';
import styled from 'styled-components';
import cookies from '@weco/common/data/cookies';
import { getCookie, setCookie } from 'cookies-next';
import useIsomorphicLayoutEffect from '../../../hooks/useIsomorphicLayoutEffect';
import { Contributor, License } from '../../../model/catalogue';

export type ApiToolbarLink = {
  id: string;
  label: string;
  value?: string;
  link?: string;
};

const ToolbarContainer = styled.div<{ mini: boolean }>`
  display: ${props => (props.mini ? 'inline-block' : 'flex')};
  background-color: ${props => props.theme.color('accent.purple')};
  color: ${props => props.theme.color('white')};
  z-index: 100;
`;

const LinkList = styled.ul.attrs({
  className: 'font-size-5',
})`
  display: flex;
  list-style: none;
  margin: 0;
`;

/** tzitzit is a tool used by the Digital Editorial team to create TASL
 * strings for Prismic. https://github.com/wellcomecollection/tzitzit
 *
 * This creates a link to a tzitzit which is pre-filled with the catalogue
 * metadata for this image.  They may not want to use it directly
 * (e.g. they may tidy up the title), but it should save them some copy/pasting
 * from the catalogue.
 *
 */

export function setTzitzitParams({
  title,
  sourceLink,
  licence,
  contributors,
}: {
  title: string;
  sourceLink: string;
  licence: License | undefined;
  contributors: Contributor[] | undefined;
}): ApiToolbarLink | undefined {
  const licenceId = licence?.id.toUpperCase();

  // We should not be using in copyright images in Stories.
  // TODO: Double check this is our policy with the Editorial team.
  if (licenceId === 'INC')
    return {
      id: 'tzitzit',
      label: '(no tzitzit, this image is in copyright)',
    };

  const params = new URLSearchParams();
  params.set('title', title);
  params.set('sourceName', 'Wellcome Collection');
  params.set('sourceLink', sourceLink);
  if (licenceId) params.set('licence', licenceId);
  if (contributors && contributors.length > 0)
    params.set('author', contributors[0].agent.label);

  return {
    id: 'tzitzit',
    label: 'tzitzit',
    link: `https://s3-eu-west-1.amazonaws.com/tzitzit.wellcomecollection.org/index.html?${params.toString()}`,
  };
}

export function createPrismicLink(prismicId: string): ApiToolbarLink {
  return {
    id: 'prismic',
    label: 'Edit in Prismic',
    link: `https://wellcomecollection.prismic.io/documents~b=working&c=published&l=en-gb/${prismicId}/`,
  };
}

function getAnchorLinkUrls() {
  // This function currently only extracts the ids from h2, h3, and h4 tags
  const getAllHeadingIds = [...document.querySelectorAll('h2, h3, h4')].map(
    item => item.id
  );
  // This function extracts any data-toolbar-anchor divs with the view to extracting the id values
  // This can be used across the codebase (where data-toolbar-anchor is used) but at the moment is only used in audio & BSL guides
  // Please note: an audio/BSL guide must contain and audio or video file with an accompanying title or no id will exist
  // and no link will be created
  const extractedAudioBSLAttributes = [
    ...document.querySelectorAll('div[data-toolbar-anchor="apiToolbar"]'),
  ].map(item => item.id);

  // Remove empty ids and then append them to the current url with # to
  // create the anchor link
  // e.g. weco.org/guides/exhibitions/YvUALRAAACMA2h8V/captions-and-transcripts#anchor-id
  const extractedHeadingIdURLs = getAllHeadingIds
    .filter(Boolean)
    .map(id => `${document.URL}#${id}`);
  const extractedAudioBSLURLs = extractedAudioBSLAttributes
    .filter(Boolean)
    .map(id => `${document.URL}#${id}`);

  const csvAsSingleColumn =
    extractedHeadingIdURLs.join('\n') + extractedAudioBSLURLs.join('\n');
  // Push the list of urls to the clipboard
  if (navigator && navigator.clipboard && navigator.clipboard.writeText)
    return navigator.clipboard.writeText(csvAsSingleColumn).then(() => {
      alert('All anchor links on this page have been copied to clipboard!');
    });
  return alert(
    'The Clipboard API is not available. No anchor links have been copied.'
  );
}

type Props = {
  links?: ApiToolbarLink[];
};

const ApiToolbar: FunctionComponent<Props> = ({ links = [] }) => {
  const [mini, setMini] = useState<boolean>(false);

  useIsomorphicLayoutEffect(() => {
    setMini(getCookie(cookies.apiToolbarMini) === true);
  }, []);

  const propValue = (prop: ApiToolbarLink) => {
    return `${prop.label}${prop.value ? ` : ${prop.value}` : ''}`;
  };

  return (
    <ToolbarContainer mini={mini}>
      <div
        style={{
          display: 'flex',
          alignItems: 'center',
          flexGrow: 1,
        }}
      >
        {!mini && (
          <>
            <span
              className="h3"
              style={{
                marginLeft: '10px',
              }}
            >
              API toolbar
            </span>
            <LinkList>
              {links.map(prop => (
                <li
                  key={prop.id}
                  style={{
                    paddingRight: '10px',
                    paddingLeft: '10px',
                    borderLeft: '1px solid #bcbab5',
                  }}
                >
                  {prop.link && <a href={prop.link}>{propValue(prop)}</a>}
                  {!prop.link && propValue(prop)}
                </li>
              ))}
            </LinkList>
          </>
        )}
      </div>
      <button
        className="plain-button"
        type="button"
        onClick={() => {
          getAnchorLinkUrls();
        }}
        style={{ padding: '10px' }}
      >
        <>‚öìÔ∏è</>
      </button>

      <button
        className="plain-button"
        type="button"
        onClick={() => {
          setMini(!mini);
          setCookie(cookies.apiToolbarMini, !mini, { path: '/' });
        }}
        style={{ padding: '10px' }}
      >
        {!mini && <>ü§è</>}
        {mini && <>üëê</>}
      </button>
    </ToolbarContainer>
  );
};

export default ApiToolbar;
