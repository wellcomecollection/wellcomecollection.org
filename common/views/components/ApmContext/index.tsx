import { ApmBase } from '@elastic/apm-rum';
import getConfig from 'next/config';
import { createContext, FunctionComponent, PropsWithChildren } from 'react';

import useApmRum from './useApmRum';

type ApmContextData = {
  apm?: ApmBase;
};

export const ApmContext = createContext<ApmContextData>({});

const { publicRuntimeConfig } = getConfig();

export const ApmContextProvider: FunctionComponent<PropsWithChildren> = ({
  children,
}) => {
  const apm = useApmRum(publicRuntimeConfig.apmConfig);
  return <ApmContext.Provider value={{ apm }}>{children}</ApmContext.Provider>;
};
