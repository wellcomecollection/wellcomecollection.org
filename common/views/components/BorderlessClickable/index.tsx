import {
  ComponentProps,
  forwardRef,
  ForwardRefRenderFunction,
  ReactNode,
  SyntheticEvent,
} from 'react';
import styled from 'styled-components';

import { IconSvg } from '@weco/common/icons';
import { classNames, font } from '@weco/common/utils/classnames';
import {
  BaseButtonInner,
  BasicButton,
  ButtonIconWrapper,
} from '@weco/common/views/components/Buttons/Buttons.styles';
import Icon from '@weco/common/views/components/Icon';

type ClickableElement = 'a' | 'button';

type StyleProps = {
  $isActive?: boolean;
};

const BorderlessClickableStyle = styled(BasicButton)<StyleProps>`
  background: transparent;
  color: ${props => props.theme.color('neutral.700')};
  padding: 10px 8px;

  &:hover {
    background: ${props => props.theme.color('neutral.300')};
  }

  ${props =>
    props.$isActive &&
    `
    background: ${props.theme.color('neutral.300')};
  `}
`;

type Props = {
  icon?: IconSvg;
  iconLeft?: IconSvg;
  text: ReactNode;
  isTextHidden?: boolean;
  isActive?: boolean;
};

type BorderlessClickableProps = Props & { as: ClickableElement };
const Button: ForwardRefRenderFunction<
  HTMLButtonElement,
  BorderlessClickableProps
> = (
  {
    as,
    icon,
    iconLeft,
    text,
    isTextHidden,
    isActive,
    ...elementProps
  }: BorderlessClickableProps,
  ref
) => {
  return (
    <BorderlessClickableStyle
      as={as}
      $isActive={isActive}
      ref={ref}
      {...elementProps}
    >
      <BaseButtonInner $isInline={true}>
        <>
          {iconLeft && (
            <ButtonIconWrapper $iconAfter={false}>
              {/* This is all a little hacky and will need some tidy up */}
              {/* We currently only use this in the header sign in button */}
              <span
                className={font('intr', 4)}
                style={{ transform: 'translateY(0.01em)' }}
              >
                <Icon icon={iconLeft} matchText={true} />
              </span>
            </ButtonIconWrapper>
          )}
          <span
            className={classNames({
              'visually-hidden': !!isTextHidden,
            })}
          >
            {text}
          </span>
          {icon && (
            <ButtonIconWrapper $iconAfter={true}>
              <Icon icon={icon} />
            </ButtonIconWrapper>
          )}
        </>
      </BaseButtonInner>
    </BorderlessClickableStyle>
  );
};

const BorderlessClickable = forwardRef<
  HTMLButtonElement,
  BorderlessClickableProps
>(Button);

type BorderlessLinkProps = Props & ComponentProps<'a'>;
const Link = (props, ref) => {
  return <BorderlessClickable as="a" ref={ref} {...props} />;
};

const BorderlessLink = forwardRef<HTMLButtonElement, BorderlessLinkProps>(Link);

type BorderlessButtonProps = Props &
  ComponentProps<'button'> & {
    clickHandler?: (event: SyntheticEvent<HTMLButtonElement>) => void;
  };
const ButtonOuter = (
  { clickHandler, ...elementProps }: BorderlessButtonProps,
  ref
) => {
  function onClick(event: SyntheticEvent<HTMLButtonElement>) {
    clickHandler && clickHandler(event);
  }

  return (
    <BorderlessClickable
      as="button"
      onClick={onClick}
      ref={ref}
      {...elementProps}
    />
  );
};

const BorderlessButton = forwardRef<HTMLButtonElement, BorderlessButtonProps>(
  ButtonOuter
);

export { BorderlessLink, BorderlessButton };
