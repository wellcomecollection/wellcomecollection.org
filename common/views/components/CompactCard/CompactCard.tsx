import {
  ComponentProps,
  FunctionComponent,
  ReactElement,
  ReactNode,
} from 'react';
import { trackEvent } from '../../../utils/ga';
import DateRange from '../DateRange/DateRange';
import EventDateRange from '../EventDateRange/EventDateRange';
import StatusIndicator from '../StatusIndicator/StatusIndicator';
import LabelsList from '../LabelsList/LabelsList';
import ImagePlaceholder from '../ImagePlaceholder/ImagePlaceholder';
import ImageType from '../Image/Image';
import { ColorSelection } from '../../../model/color-selections';
import MediaObjectBase from '../MediaObjectBase/MediaObjectBase';

type Props = {
  url: string | null;
  title: string;
  labels: ComponentProps<typeof LabelsList>;
  description: string | ReactElement | null;
  urlOverride: string | null;
  extraClasses?: string;
  partNumber: number | null;
  color: ColorSelection | null;
  Image: ReactElement<typeof ImageType | typeof ImagePlaceholder> | null;
  DateInfo:
    | ReactElement<typeof DateRange>
    | ReactElement<typeof EventDateRange>
    | null;
  StatusIndicator: ReactElement<typeof StatusIndicator> | null;
  ExtraInfo?: ReactNode | null;
  xOfY: { x: number; y: number };
  OverrideImageWrapper?: ReactNode;
  OverrideTextWrapper?: ReactNode;
  OverrideTitleWrapper?: ReactNode;
};

const CompactCard: FunctionComponent<Props> = ({
  url,
  title,
  labels,
  description,
  urlOverride,
  extraClasses,
  partNumber,
  color,
  Image,
  DateInfo,
  StatusIndicator,
  ExtraInfo,
  xOfY,
  OverrideImageWrapper,
  OverrideTextWrapper,
  OverrideTitleWrapper,
}: Props): ReactElement<Props> => {
  return (
    <MediaObjectBase
      url={url}
      title={title}
      Image={Image}
      partNumber={partNumber}
      color={color}
      StatusIndicator={StatusIndicator}
      description={description}
      urlOverride={urlOverride}
      extraClasses={extraClasses}
      DateInfo={DateInfo}
      ExtraInfo={ExtraInfo}
      labels={labels}
      xOfY={xOfY}
      OverrideImageWrapper={OverrideImageWrapper}
      OverrideTextWrapper={OverrideTextWrapper}
      OverrideTitleWrapper={OverrideTitleWrapper}
      onClick={(): void => {
        trackEvent({
          category: 'CompactCard',
          action: 'follow link',
          label: title,
        });
      }}
    />
  );
};

export default CompactCard;
