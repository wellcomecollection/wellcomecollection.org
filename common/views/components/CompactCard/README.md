## Purpose

[Edit this on GitHub](https://github.com/wellcomecollection/wellcomecollection.org/edit/master/common/views/components/CompactCard/README.md)
