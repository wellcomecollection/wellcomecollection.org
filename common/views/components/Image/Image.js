// @flow
import {conditionalClassNames} from '../../../utils/classnames';
import {convertImageUri} from '../../../utils/convert-image-uri';
import {imageSizes} from '../../../utils/image-sizes';
import {Fragment} from 'react';

export type Props = {|
  contentUrl: string,
  width: number,
  alt: string,
  height?: number,
  clipPathClass?: ?string,
  caption?: string,
  lazyload?: boolean,
  sizesQueries?: string,
  copyright?: string,
  defaultSize?: number,
  clickHandler?: () => void,
  zoomable?: boolean,
  extraClasses?: string,
  style?: { [string]: any } // TODO: find flowtype for this
|}

const Image = (props: Props) => (
  <Fragment>
    <noscript dangerouslySetInnerHTML={{__html: `
      <img width='${props.width}'
        height='${props.height || ''}'
        className='image image--noscript'
        src='${convertImageUri(props.contentUrl, 640, false)}'
        alt='${props.alt || ''}' />`}} />

    {props.clipPathClass ? (
      <Fragment>
        <Img {...props} clipPathClass={null} />
        <Img {...props} />
      </Fragment>
    ) : <Img {...props} />}
  </Fragment>
);

const Img = ({
  width,
  height,
  contentUrl,
  clipPathClass,
  caption,
  copyright,
  lazyload = true,
  sizesQueries = '100vw',
  defaultSize = 30,
  alt = '',
  clickHandler,
  zoomable,
  extraClasses,
  style
}: Props) => {
  const sizes = imageSizes(width);
  return (
    <img width={width}
      height={height}
      className={conditionalClassNames({
        'image': true,
        'lazy-image lazyload': lazyload,
        'cursor-zoom-in': Boolean(zoomable),
        [`promo__image-mask ${clipPathClass || ''}`]: clipPathClass,
        [`${extraClasses || ''}`]: Boolean(extraClasses)
      })}
      src={convertImageUri(contentUrl, defaultSize, false)}
      data-srcset={sizes.map(size => {
        return `${convertImageUri(contentUrl, size, false)} ${size}w`;
      })}
      sizes={sizesQueries}
      data-copyright={copyright}
      onClick={clickHandler}
      alt={alt || ''}
      style={style} />
  );
};

export default Image;
