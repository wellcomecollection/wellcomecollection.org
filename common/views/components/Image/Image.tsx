import { classNames } from '../../../utils/classnames';
import { convertPrismicImageUri } from '../../../utils/convert-image-uri';
import { imageSizes, supportedSizes } from '../../../utils/image-sizes';
import { isNotUndefined } from '../../../utils/array';
import { FunctionComponent } from 'react';

export type Props = {
  contentUrl: string;
  width?: number;
  height?: number;
  alt: string | null;
  lazyload?: boolean;
  sizesQueries?: string;
  copyright?: string;
  defaultSize?: number;
  clickHandler?: () => void;
  zoomable?: boolean;
  extraClasses?: string;
  style?: Record<string, string>;
  srcsetRequired?: boolean;
};

const Img = ({
  width,
  height,
  alt = '',
  contentUrl,
  lazyload = true,
  sizesQueries = '100vw',
  copyright,
  defaultSize = 30,
  clickHandler,
  zoomable,
  extraClasses,
  style,
  srcsetRequired = true,
}: Props) => {
  const sizes = isNotUndefined(width) ? imageSizes(width) : supportedSizes;
  return (
    <img
      width={width}
      height={height}
      className={classNames({
        image: true,
        'bg-charcoal font-white': true,
        'lazy-image lazyload': lazyload,
        'cursor-zoom-in': Boolean(zoomable),
        [`${extraClasses || ''}`]: Boolean(extraClasses),
      })}
      src={convertPrismicImageUri(contentUrl, defaultSize)}
      data-srcset={
        srcsetRequired
          ? sizes.map(size => {
              return `${convertPrismicImageUri(contentUrl, size)} ${size}w`;
            })
          : undefined
      }
      sizes={srcsetRequired ? sizesQueries : undefined}
      data-copyright={copyright}
      onClick={clickHandler}
      alt={alt || ''}
      style={style}
    />
  );
};

const Image: FunctionComponent<Props> = (props: Props) => {
  const classes = classNames({
    'image image--noscript bg-charcoal font-white': true,
    [`${props.extraClasses || ''}`]: Boolean(props.extraClasses),
  });
  return (
    <>
      <noscript
        dangerouslySetInnerHTML={{
          __html: `
      <img width='${props.width || ''}'
        height='${props.height || ''}'
        class='${classes}'
        src='${convertPrismicImageUri(props.contentUrl, 640)}'
        alt='${props.alt || ''}' />`,
        }}
      />
      <Img {...props} />
    </>
  );
};

export default Image;
