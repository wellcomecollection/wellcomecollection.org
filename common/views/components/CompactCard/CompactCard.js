// @flow
import type {Element, ElementProps} from 'react';
import {grid, font, spacing, conditionalClassNames} from '../../../utils/classnames';
import DateRange from '../DateRange/DateRange';
import StatusIndicator from '../StatusIndicator/StatusIndicator';
import LabelsList from '../LabelsList/LabelsList';
import ImagePlaceholder from '../ImagePlaceholder/ImagePlaceholder';
import PartNumberIndicator from '../PartNumberIndicator/PartNumberIndicator';
import {default as ImageType} from '../Image/Image';
import type {ColorSelection} from '../../../model/color-selections';

type Props = {|
  url: ?string,
  title: string,
  promoType: string,
  labels: ElementProps<typeof LabelsList>,
  description: ?string,
  urlOverride: ?string,
  extraClasses?: string,
  partNumber: ?number,
  color: ?ColorSelection,
  Image: ?Element<typeof ImageType | typeof ImagePlaceholder>,
  DateInfo: ?Element<typeof DateRange>,
  StatusIndicator: ?Element<typeof StatusIndicator>
|}

const CompactCard = ({
  url,
  title,
  promoType,
  labels,
  description,
  urlOverride,
  extraClasses,
  partNumber,
  color,
  Image,
  DateInfo,
  StatusIndicator
}: Props) => {
  const textGridSizes = Image
    ? {s: 7, m: 7, l: 8, xl: 8}
    : {s: 12, m: 12, l: 12, xl: 12};

  const Tag = url ? 'a' : 'div';
  return (
    <Tag
      data-component={promoType}
      data-track-event={JSON.stringify({
        category: 'component',
        action: `${promoType}:click`
      })}
      href={urlOverride || url}
      className={conditionalClassNames({
        'grid plain-link': true,
        [spacing({s: 3}, {padding: ['bottom', 'top']})]: true,
        [extraClasses || '']: Boolean(extraClasses)
      })}>
      {labels.labels.length > 0 &&
        <div className={conditionalClassNames({
          [grid({s: 12, m: 12, l: 12, xl: 12})]: true,
          [spacing({s: 1}, {margin: ['bottom']})]: true
        })}>
          <LabelsList {...labels} />
        </div>
      }
      {Image &&
        <div className={grid({s: 5, m: 5, l: 4, xl: 4})}>
          {Image}
        </div>
      }
      <div className={grid(textGridSizes)}>
        {partNumber && <PartNumberIndicator number={partNumber} color={color} />}
        <div className={`${font({s: 'WB5'})} ${spacing({s: 0}, {margin: ['top']})}`}>
          {title}
        </div>
        {DateRange &&
          <div>
            {DateInfo}
          </div>
        }
        {StatusIndicator}
        {description &&
        <div className='spaced-text'>
          <p className={font({s: 'HNL4'})}>
            {description}
          </p>
        </div>
        }
      </div>
    </Tag>
  );
};

export default CompactCard;
