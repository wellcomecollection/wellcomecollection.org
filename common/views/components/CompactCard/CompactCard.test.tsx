import CompactCard from './CompactCard';
import { mountWithTheme } from '../../../test/fixtures/enzyme-helpers';
import { mockData } from '../../../test/fixtures/components/compact-card';
import Image from '../Image/Image';
import * as ga from '../../../utils/ga';

const extraClass = 'my_extra_extra_class';

describe('CompactCard', () => {
  const componentWithImage = mountWithTheme(
    <CompactCard
      title={mockData.title}
      Image={<Image {...mockData.image.crops.square} />}
      description={mockData.text}
      primaryLabels={[]}
      secondaryLabels={[]}
      extraClasses={extraClass}
    />
  );

  it('should expect ga trackEvent tobe called ', () => {
    const spyOnTrackEvent = jest.spyOn(ga, 'trackEvent');
    componentWithImage.simulate('click');
    expect(spyOnTrackEvent).toBeCalled();
    expect(spyOnTrackEvent).toHaveBeenCalledTimes(1);
    expect(spyOnTrackEvent).toBeCalledWith({
      action: 'follow link',
      category: 'CompactCard',
      label: 'mock title',
    });
  });
});
