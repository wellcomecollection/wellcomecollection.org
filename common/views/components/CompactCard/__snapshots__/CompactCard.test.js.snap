// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompactCard should match snapshots 1`] = `
"<div class=\\"Space-sc-4az8w3-0 iRLDmT grid my_extra_extra_class\\"><div class=\\"MediaObjectBase__BaseImageWrapper-w8f9sz-0 grid__cell grid__cell--s3 grid__cell--m3 grid__cell--l3 grid__cell--xl3\\"><noscript>
      <img width='3200'
        height='3200'
        class='image image--noscript bg-charcoal font-white'
        src='https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&rect=397%2C0%2C2836%2C2836&w=640&h=640'
        alt='Two St. Bernard dogs with an avalanche victim, one tries to revive him while the other alerts the rescue party.' /></noscript><img width=\\"3200\\" height=\\"3200\\" class=\\"image bg-charcoal font-white lazy-image lazyload\\" src=\\"https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=30&amp;h=30\\" data-srcset=\\"https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=160&amp;h=160 160w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=180&amp;h=180 180w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=282&amp;h=282 282w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=320&amp;h=320 320w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=420&amp;h=420 420w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=600&amp;h=600 600w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=880&amp;h=880 880w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=960&amp;h=960 960w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=1024&amp;h=1024 1024w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=1338&amp;h=1338 1338w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=2048&amp;h=2048 2048w\\" sizes=\\"100vw\\" alt=\\"Two St. Bernard dogs with an avalanche victim, one tries to revive him while the other alerts the rescue party.\\"></div><div class=\\"MediaObjectBase__BaseTextWrapper-w8f9sz-2 grid__cell grid__cell--s9 grid__cell--m9 grid__cell--l9 grid__cell--xl9\\"><div class=\\"MediaObjectBase__BaseTitleWrapper-w8f9sz-1 card-link__title font-wb font-size-3 \\">mock title</div><div class=\\"spaced-text\\"><p class=\\"font-hnl font-size-5 \\">some text</p></div></div></div>"
`;
